pheno_utils/pheno_loader.py

```diff
@@ -119,7 +119,9 @@ def load_sample_data(
         parent_bulk: Union[None, str] = None,
         load_func: callable = None,
         concat: bool = True,
-        pivot=None, **kwargs
+        pivot=None, 
+        keep_undefined: bool = False,
```

Comment:
the arg name is a little obscure. I understand that being more verbose like `keep_undefined_research_stage` is annoying, but it's hard to guess what it means.

one more thing, consider using a class property to define this (when initializing PhenoLoader), like we've done in the past (see flexible_field_search). do you expect that people will want to switch it on or off often? if so, keep as a regular arg to the functions as well. but if you want it to be set once during init, maybe this is also fine.

Reply:
it occurred to me that sometimes, perhaps in rare occasions, we may want to merge data that has undefined research stage in a different way altogether: ignore the research stage completely (merge each participant row on the left with all participant rows on the right). the only example that I could think of right now is genetics, and it doesn't have very interesting tabular data, so it doesn't matter. also, we are not obligated to support every possible scenario in the loader (people can always write their own code). just a thought to air out, though.

Reply:
I do think that people will want to make changes in some cases during their work, so I'll add both options.

I understand the comment about the different way of merging, but I feel that this might create another loophole.
I'll open another ticket on this so we wouldn't forget about it and get to it a little bit later